src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,776 @@
+				TreeViewNode draggedNode = treeViewList.DraggedTreeViewNode;
+
+				if (draggedNode != null && treeView.CanReorderItems)
+				{
+					if (treeViewList.IsMutiSelectWithSelectedItems)
+					{
+						if (treeViewList.IsSelected(draggedOverNode))
+						{
+							args.AcceptedOperation = DataPackageOperation.None;
+							treeViewList.SetDraggedOverItem(null);
+						}
+						else
+						{
+							args.AcceptedOperation = DataPackageOperation.Move;
+							treeViewList.SetDraggedOverItem(draggedOverItem);
+						}
+					}
+					else
+					{
+						TreeViewNode walkNode = draggedOverNode.Parent;
+						while (walkNode != null && walkNode != draggedNode)
+						{
+							walkNode = walkNode.Parent;
+						}
+
+						if (walkNode != draggedNode && draggedNode != draggedOverNode)
+						{
+							args.AcceptedOperation = DataPackageOperation.Move;
+							treeViewList.SetDraggedOverItem(draggedOverItem);
+						}
+
+						treeViewList.UpdateDropTargetDropEffect(false, false, null);
+					}
+				}
+			}
+
+			base.OnDragOver(args);
+		}
+
+		protected override void OnDragEnter(Windows.UI.Xaml.DragEventArgs args)
+		{
+			TreeViewItem draggedOverItem = this;
+
+			args.AcceptedOperation = DataPackageOperation.None;
+			args.DragUIOverride.IsGlyphVisible = true;
+
+			var treeView = AncestorTreeView;
+			if (treeView != null && treeView.CanReorderItems && !args.Handled)
+			{
+				var treeViewList = treeView.ListControl;
+				TreeViewNode draggedNode = treeViewList.DraggedTreeViewNode;
+				if (draggedNode != null)
+				{
+					TreeViewNode draggedOverNode = treeView.NodeFromContainer(draggedOverItem);
+					TreeViewNode walkNode = draggedOverNode.Parent;
+
+					while (walkNode != null && walkNode != draggedNode)
+					{
+						walkNode = walkNode.Parent;
+					}
+
+					if (walkNode != draggedNode && draggedNode != draggedOverNode)
+					{
+						args.AcceptedOperation = DataPackageOperation.Move;
+					}
+
+					TreeViewNode droppedOnNode = TreeNode;
+					if (droppedOnNode != draggedNode) // Don't expand if drag hovering on itself.
+					{
+						// Set up timer.
+						if (!draggedOverNode.IsExpanded && draggedOverNode.HasChildren)
+						{
+							if (m_expandContentTimer != null)
+							{
+								var expandContentTimer = m_expandContentTimer;
+								expandContentTimer.Stop();
+								expandContentTimer.Start();
+							}
+							else
+							{
+								// Initialize timer.
+								TimeSpan interval = new TimeSpan(c_dragOverInterval);
+								var expandContentTimer = new DispatcherTimer();
+								m_expandContentTimer = expandContentTimer;
+								expandContentTimer.Interval = interval;
+								expandContentTimer.Tick += OnExpandContentTimerTick;
+								expandContentTimer.Start();
+							}
+						}
+					}
+				}
+			}
+
+			base.OnDragEnter(args);
+		}
+
+		protected override void OnDragLeave(Windows.UI.Xaml.DragEventArgs args)
+		{
+			if (!args.Handled)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'args' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)